[BC-Breaking]Remove capture_pre_autograd_graph references in quantization

yushangdi opened the pull request:
Summary:
As title

This is a BC-breaking change because graph produced by "capture_pre_autograd_graph" cannot be input to quantization anymore. But this is ok, since this API is deprecated for a while and is going to be deleted. We have removed all call sites of it. 


We remove the deprecated API references in code, docs, and tests.


We also removed two tests that specific to capture_pre_autograd_graph API.

Test Plan: CI

Differential Revision: D65351887




cc @ezyang @gchanan

jerryzh168 commented:
cc @andrewor14 can you follow up on how does the change of training ir affect batchnorm op consolidation work?

yushangdi commented:
There are some test failures because a call to `capture_pre_autograd_graph` is recently added in https://github.com/pytorch/pytorch/pull/139054. I'd have to wait until it's imported internally to overwrite it.

andrewor14 commented:
> cc @andrewor14 can you follow up on how does the change of training ir affect batchnorm op consolidation work?

Sorry just saw this. It actually makes batch norm consolidation much lower priority, because export no longer produces the same batch norm ops as before. In particular, now we get `aten.batch_norm` directly in the graph, which is kind of already the "consolidated" op from export perspective. Users won't see `aten.cudnn_batch_norm` or `aten._native_batch_norm_legit` anymore if they use `export_for_training`

yushangdi commented:
@andrewor14 @jerryzh168 We've removed all "capture_pre_autograd_graph" call sites now, and we are finally deleting it. Can I get a review for this PR? Thanks!!

andrewor14 commented on torch/ao/quantization/pt2e/utils.py:
```diff
@@ -381,10 +380,9 @@ def _get_aten_graph_module_for_pattern(
-            kwargs,
+        raise RuntimeError(
+            "capture_pre_autograd_graph is deprecated and will be deleted soon."
+            "Please use torch.export.export_for_training instead."
```
This is a BC-breaking change right? Should we mark it as such in the PR title and description?

yushangdi replied:
> This is a BC-breaking change right? Should we mark it as such in the PR title and description?

sure!

andrewor14 reviewed (approved):
Looks great, thanks! Please also import this to make sure no existing use cases are being affected.

jerryzh168 reviewed (approved):
looks good, @andrewor14 can you follow up on the batchnorm and qat related things? (maybe make a post?) I'm still not clear on how the capture changes these things

yushangdi committed:
Remove capture_pre_autograd_graph references in quantization (#139505)

Summary:

As title

We remove the deprecated API references in code, docs, and tests.


We also removed two tests that specific to capture_pre_autograd_graph API.

Test Plan: CI

Reviewed By: andrewor14, tugsbayasgalan

Differential Revision: D65351887

yushangdi commented:
@pytorchbot merge -f landed internally

yushangdi commented:
@pytorchbot merge -f "landed internally"

yushangdi commented:
@pytorchbot merge -f "landed internally"
